tools/storyteller/StoryWriter.java

```diff
@@ -1,88 +1,88 @@
 import java.awt.Toolkit;
 import java.awt.image.BufferedImage;
-import java.io.File;
 import java.io.IOException;
 import java.nio.file.Paths;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 import javax.imageio.ImageIO;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import tools.storyteller.Protos.Config;
-import tools.storyteller.Protos.FileData;
-import tools.storyteller.Protos.ScreenshotMetadata;
+import tools.storyteller.service.Protos.Story;
+import tools.storyteller.service.Protos.StoryItem;
+import tools.storyteller.service.Protos.StoryList;
 
 
 /* Storyteller writer that writes story files. */
 @Singleton
 public class StoryWriter {
   private static final Logger log = Logger.getForClass();
-  public static DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
+  private static final DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
 
-  private Config config;
-  private FileUtils fileUtils;
+  private final Config config;
+  private final FileUtils fileUtils;
+  private Story.Builder currentStoryBuilder;
+  private List<Story> allStories;
 
   @Inject
   public StoryWriter(StorytellerConfig storytellerConfig, FileUtils fileUtils) {
     this.config = storytellerConfig.getConfig();
     this.fileUtils = fileUtils;
+    allStories = new ArrayList<>();
   }
 
-  private String getSharedStoriesPath() {
-    return Storyteller.getSharedStoriesPath(config);
+  void startStory(String project) {
+    currentStoryBuilder = Story.newBuilder();
+    currentStoryBuilder
+        .setStartTimeMs(getCurrentTimestamp())
+        .setProject(project);
+    allStories.add(currentStoryBuilder.build());
+    saveStories();
   }
 
-  private String getUnsharedStoriesPath() {
-    return Storyteller.getUnsharedStoriesPath(config);
-  }
-
-  public void writeStatusFile(FileData.Type type, Message contents) {
-    fileUtils.mkdirs(getUnsharedStoriesPath());
-    // We shift by a second to screenshots will be within START and END.
-    int secondsShift = 0;
-    if (type == FileData.Type.START) {
-      secondsShift = -1;
-    }
-    if (type == FileData.Type.END) {
-      secondsShift = 1;
-    }
-
-    String timeString = getCurrentTimeString(secondsShift);
-    String filename = timeString + "." + type;
-    try {
-      File statusFile = Paths.get(getUnsharedStoriesPath(), filename).toFile();
-      if (contents == null) {
-        statusFile.createNewFile();
-      } else {
-        fileUtils.writePrototxt(contents, statusFile.toString());
-      }
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-    if (type != FileData.Type.RUNNING) {
-      System.out.println(String.format("\n%s tracking: %s", type, timeString));
+  void updateStory(String project) {
+    if(currentStoryBuilder == null){
+      throw new RuntimeException("'currentStoryBuilder' is not initialized");
```

Comment:
s/RuntimeException/IllegalStateException
IllegalStateException inherits from RuntimeException, but is a better fit.